src/benchmarklib/tpcc/tpcc_table_generator.hpp

```diff
@@ -38,40 +39,168 @@ class TpccTableGenerator : public opossum::AbstractBenchmarkTableGenerator {
 
   std::shared_ptr<Table> generate_history_table();
 
-  typedef std::vector<std::vector<std::vector<size_t>>> order_line_counts_type;
+  using OrderLineCounts = std::vector<std::vector<std::vector<size_t>>>;
 
-  order_line_counts_type generate_order_line_counts();
+  OrderLineCounts generate_order_line_counts();
 
-  std::shared_ptr<Table> generate_order_table(const order_line_counts_type& order_line_counts);
+  std::shared_ptr<Table> generate_order_table(const OrderLineCounts& order_line_counts);
 
-  std::shared_ptr<Table> generate_order_line_table(const order_line_counts_type& order_line_counts);
+  std::shared_ptr<Table> generate_order_line_table(const OrderLineCounts& order_line_counts);
 
   std::shared_ptr<Table> generate_new_order_table();
 
-  std::map<std::string, std::shared_ptr<Table>> generate_all_tables() override;
+  std::unordered_map<std::string, BenchmarkTableInfo> generate() override;
 
-  static TpccTableGeneratorFunctions table_generator_functions();
-
-  std::shared_ptr<Table> generate_table(const std::string& table_name);
-
-  const size_t _warehouse_size;
+  const size_t _num_warehouses;
   const time_t _current_date = std::time(nullptr);
-  const EncodingConfig _encoding_config;
 
  protected:
-  void _encode_table(const std::string& table_name, const std::shared_ptr<Table>& table);
-
   template <typename T>
-  std::vector<T> _generate_inner_order_line_column(std::vector<size_t> indices,
-                                                   order_line_counts_type order_line_counts,
+  std::vector<T> _generate_inner_order_line_column(std::vector<size_t> indices, OrderLineCounts order_line_counts,
                                                    const std::function<T(std::vector<size_t>)>& generator_function);
 
   template <typename T>
   void _add_order_line_column(std::vector<Segments>& segments_by_chunk, TableColumnDefinitions& column_definitions,
                               std::string name, std::shared_ptr<std::vector<size_t>> cardinalities,
-                              order_line_counts_type order_line_counts,
+                              OrderLineCounts order_line_counts,
                               const std::function<T(std::vector<size_t>)>& generator_function);
 
-  TpccRandomGenerator _random_gen;
+  TPCCRandomGenerator _random_gen;
+
+  /**
```

Comment:
From here until the end, this was moved from the AbstractBenchmarkTableGenerator